binance_margin.go

```diff
@@ -0,0 +1,258 @@
+package binance
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+const (
+	listenKeyTypeMarginAPI         = "sapi/v1/userDataStream"
+	listenKeyTypeIsolatedMarginAPI = "sapi/v1/userDataStream/isolated"
+)
+
+// CreateListenKeyMargin create a listen key for user data stream
+func (bc *Client) CreateListenKeyMargin() (string, error) {
+	return bc.createListenKey(listenKeyTypeMarginAPI)
+}
+
+// KeepListenKeyAliveMargin keep it alive
+func (bc *Client) KeepListenKeyAliveMargin(listenKey string) error {
+	return bc.keepListenKeyAlive(listenKey, listenKeyTypeMarginAPI)
+}
+
+// CreateListenKeyIsolatedMargin create a listen key for user data stream
+func (bc *Client) CreateListenKeyIsolatedMargin() (string, error) {
+	return bc.createListenKey(listenKeyTypeIsolatedMarginAPI)
+}
+
+// KeepListenKeyAliveIsolatedMargin keep it alive
+func (bc *Client) KeepListenKeyAliveIsolatedMargin(listenKey string) error {
+	return bc.keepListenKeyAlive(listenKey, listenKeyTypeIsolatedMarginAPI)
+}
+
+type marginCommonResult struct {
+	TranID uint64 `json:"tranId"`
+}
+
+func (bc *Client) CrossMarginTransfer(asset, amount string, spotToMargin bool) (uint64, *FwdData, error) {
+	transType := ""
+	if spotToMargin {
+		transType = "1"
+	} else {
+		transType = "2"
+	}
+	var (
+		result marginCommonResult
+	)
+	requestURL := fmt.Sprintf("%s/sapi/v1/margin/transfer", apiBaseURL)
+	req, err := NewRequestBuilder(http.MethodPost, requestURL, nil)
+	if err != nil {
+		return 0, nil, err
+	}
+	rr := req.WithHeader(apiKeyHeader, bc.apiKey).
+		WithParam("asset", asset).
+		WithParam("amount", amount).
+		WithParam("type", transType).
+		SignedRequest(bc.secretKey)
+	fwd, err := bc.doRequest(rr, &result)
+	if err != nil {
+		return 0, fwd, err
+	}
+
+	return result.TranID, fwd, err
+}
+
+func (bc *Client) CrossMarginBorrow(asset, symbol, amount string, isIsolated bool) (uint64, *FwdData, error) {
+	isISO := ""
```

Comment:
this could be set as "FALSE" at first place. Please look at my comment in CrossMarginRepay.